traits/ctraits.c

```diff
@@ -310,9 +310,16 @@ fatal_trait_error ( void ) {
 +----------------------------------------------------------------------------*/
 
 static int
-invalid_attribute_error ( void ) {
+invalid_attribute_error ( PyObject * name ) {
 
-    PyErr_SetString( PyExc_TypeError, "attribute name must be string" );
+    // Build error message by collecting the type of the object received
+    PyTypeObject* obj_type = name->ob_type;
+    const char* type_name = obj_type->tp_name;
+    char err_msg[60] = "attribute name must be string. Got ";
+
+    strncat(err_msg, type_name, 20);
+
+    PyErr_SetString( PyExc_TypeError, err_msg );
```

Comment:
Thanks @jvkersch for the feedback. I pushed an updated version that uses your suggestion and adds a `repr` of the value for even more feedback. More thoughts?